lib/r10k/git/rugged/credentials.rb

```diff
@@ -125,4 +143,49 @@ def get_git_username(url, username_from_url)
 
     user
   end
+
+  def github_app_token(app_id, private_key, ttl)
+    raise R10K::Git::GitError, _('App id contains invalid characters.') unless app_id =~ /^\d+$/
+    raise R10K::Git::GitError, _('Token ttl contains invalid characters.') unless ttl =~ /^\d+$/
+    raise R10K::Git::GitError, _('App key is missing or unreadable') unless File.readable?(private_key)
+
+    begin
+      ssl_key = OpenSSL::PKey::RSA.new(File.read(private_key).strip)
+      ssl_key.private?
```

Comment:
This just returns a boolean, right? Should this be part of a conditional or something? Not sure what it's doing here, I don't think `false` will take us to the `rescue`..